lib/logidze/model.rb

```diff
@@ -117,6 +121,18 @@ def switch_to!(version)
       self.class.without_logging { save! }
     end
 
+    def association(name)
+      association = super
+
+      if logidze_past?
+        association.singleton_class.prepend Logidze::VersionedAssociation
+      end
+
+      association.reload if @logidze_requested_ts
```

Comment:
Yes, it was pretty brutal :) I used `reload` to invoke `load_target` in case of already loaded before `at` invocation `belongs_to` association:
```
article.user
old_article = article.at(...)
old_article.user # here load_target will not be called, so we won't have versioned user
```

But I can work around this issue another way, without `reload` or `reset` at all. I pushed commit recently with these changes.

But it seems, that there is no clean way to achieve `size` method working, so I got back to `length` so far